client/factory.go

```diff
@@ -85,8 +75,8 @@ func NewFactory(config FactoryConfig) (*Factory, error) {
 // GetDeploymentsClient returns DeploymentsClient that is used for management of deployments and
 // ARM templates. The client (for specified cluster) is cached after creation, so the same client
 // is returned every time.
-func (f *Factory) GetDeploymentsClient(cr v1alpha1.AzureConfig) (*resources.DeploymentsClient, error) {
-	client, err := f.getClient(cr, "DeploymentsClient", newDeploymentsClient)
+func (f *Factory) GetDeploymentsClient(credentialNamespace, credentialName string) (*resources.DeploymentsClient, error) {
```

Comment:
Instead of passing a CR, we pass the namespace and name for the credentials that we want to use. This makes the factory CR independent.

Reply:
> This makes the factory CR independent. Do you think it makes sense?

It does.

> is it awkward?

No. :) 

WDYT about having a single parameter, maybe something like `credentialKey`, `credentialConfig`, `credentialID`, so it's more neutral? (then you can slice it and dice it in the provider code as you wish).

I'm thinking that getting credential by namespace and name is maybe too specific for it to be neutral. On the other hand if some other hypothetical credential provider needs only one string, then one of the two parameters would simply be ignored.

Reply:
Talked over slack. Creating such parameter would make the clients of the provider having to know how to _encode_ the namespace and name of the credential. Furthermore we only have the k8s provider anway, so let's not overengineer it.